### ISSUE TITLE
Code Issue

### ISSUE THREAD

**Issue by WJLee123** [WJLee123/Test123#1]
`
import urllib
import os
import zipfile
import pandas as pd
import tensorflow as tf

def download_and_extract_data():
    url = 'https://storage.googleapis.com/download.tensorflow.org/data/certificate/household_power.zip'
    urllib.request.urlretrieve(url, 'household_power.zip')
    with zipfile.ZipFile('household_power.zip', 'r') as zip_ref:
        zip_ref.extractall()

def normalize_series(data, min, max):
    data = data - min
    data = data / max
    return data

def windowed_dataset(series, batch_size, n_past=24, n_future=24, shift=1):
    ds = tf.data.Dataset.from_tensor_slices(series)
    ds = ds.window(size = 1,
                   shift = shift,
                           drop_remainder = True)
    ds = ds.flat_map(lambda w: w.batch(n_past + n_future))
    ds = ds.map(lambda w: (w[:-1], w[1:]))
    return ds.batch(batch_size).prefetch(1)

def solution_model():
    download_and_extract_data()
    df = pd.read_csv('household_power_consumption.csv', sep=',',
                     infer_datetime_format=True, index_col='datetime', header=0)
    N_FEATURES = len(df.columns)

    data = df.values
    split_time = int(len(data) * 0.5)
    data = normalize_series(data, data.min(axis=0), data.max(axis=0))

    x_train = data[:split_time]
    x_valid = data[split_time:]

    BATCH_SIZE = 32  # ADVISED NOT TO CHANGE THIS
    N_PAST = 24  # DO NOT CHANGE THIS
    N_FUTURE = 24  # DO NOT CHANGE THIS
    SHIFT = 1  # DO NOT CHANGE THIS
    train_set = windowed_dataset(series=x_train, batch_size=BATCH_SIZE,
                                 n_past=N_PAST, n_future=N_FUTURE,
                                 shift=SHIFT)
    valid_set = windowed_dataset(series=x_valid, batch_size=BATCH_SIZE,
                                 n_past=N_PAST, n_future=N_FUTURE,
                                 shift=SHIFT)

    # Code to define your model.
    model = tf.keras.models.Sequential([
        tf.keras.layers.LSTM(32,activation='tanh',return_sequences=True),
        tf.keras.layers.Dense(N_FEATURES)
    ])

    model.summary()

    # Code to train and compile the model
    optimizer =  tf.keras.optimizers.SGD(lr=1e-8, momentum=0.9)
    model.compile(
        # YOUR CODE HERE
        loss=tf.keras.losses.Huber(),
        optimizer=optimizer,
        metrics=["mae"]
    )
    model.fit(
        # YOUR CODE HERE
        train_set, epochs=100
    )

    return model

if __name__ == '__main__':
    model = solution_model()
    model.save("model.h5")
`
